spec/ibge/reader/base_spec.rb

```diff
@@ -1,47 +1,93 @@
-    it { should respond_to :columns }
+    describe '.columns' do
+      it { should respond_to :columns }
 
-    it 'maps the column index to a recognized column name' do
-      expect(subject.defined_columns).to eq "0": :code
+      it 'maps the column index to a recognized column name' do
+        expect(subject.defined_columns).to eq sidra: { "0": :code }
+      end
     end
-  end
 
-  describe '.read' do
-    context 'without pass a :filename as option' do
-      it 'reads the file inside data folder' do
-        expect(subject.read).to include code: "11"
+    describe '.read' do
+      context 'without pass a :filename as option' do
+        it 'reads the file inside data folder' do
+          expect(subject.read).to include code: "11"
+        end
+      end
+
+      context 'passing a :filename' do
+        it 'reads the file provided' do
+          filename = "spec/fixtures/DTB_example_subdistrito.xls"
+
+          expect(subject.read(filename: filename)).to include code: "11"
+        end
       end
     end
 
-    context 'passing a :filename' do
-      it 'reads the file provided' do
-        filename = "spec/fixtures/DTB_example_subdistrito.xls"
+    describe '.selected_source' do
+      it 'show the default file source to read' do
+        expect(subject.selected_source).to eq :sidra
+      end
+    end
 
-        expect(subject.read(filename: filename)).to include code: "11"
+    describe '#filename' do
+      it 'sets the default to file inside data folder' do
+        expect(subject.new.filename).to include 'data/sidra.xls'
       end
     end
   end
 
-  describe '.selected_source' do
-    it 'show the default file source to read' do
-      expect(subject.selected_source).to eq :sidra
+  context 'without source defined' do
+    class ReaderWithoutSource < Ibge::Reader::Base
+      columns "0": :code
     end
-  end
 
-  describe '#filename' do
-    it 'sets the default to file inside data folder' do
-      expect(subject.new.filename).to include 'data/sidra.xls'
+    subject { ReaderWithoutSource }
+
+    describe '.columns' do
+      it { should respond_to :columns }
+
+      it 'maps the column index to a recognized column name' do
+        expect(subject.defined_columns).to eq dtb: { "0": :code }
+      end
+    end
+
+    describe '.read' do
+      context 'without pass a :filename as option' do
+        it 'reads the file inside data folder' do
+          expect(subject.read).to include code: "11"
+        end
+      end
+
+      context 'passing a :filename' do
+        it 'reads the file provided' do
+          filename = "spec/fixtures/DTB_example_subdistrito.xls"
+
+          expect(subject.read(filename: filename)).to include code: "11"
+        end
+      end
+    end
+
+    describe '.selected_source' do
+      it 'show the default file source to read' do
+        expect(subject.selected_source).to eq :dtb
+      end
+    end
+
+    describe '#filename' do
+      it 'sets the default to file inside data folder' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.